pallets/teerex/src/lib.rs

```diff
@@ -122,6 +123,15 @@ pub mod pallet {
 	#[pallet::getter(fn enclave_count)]
 	pub type EnclaveCount<T: Config> = StorageValue<_, u64, ValueQuery>;
 
+	#[pallet::type_value]
+	pub fn DefaultEnclaveCountMax() -> u64 {
+		MAX_ENCLAVE_NUMBER
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn enclave_count_max)]
+	pub type EnclaveCountMax<T: Config> = StorageValue<_, u64, ValueQuery, DefaultEnclaveCountMax>;
```

Comment:
I think normally it's called `MaxEnclaveCount` (means we are setting a maximum value)